src/client.js

```diff
@@ -158,16 +162,20 @@ export class StitchClient {
    * @param {Object} [options] additional authentication options
    * @returns {Promise} which resolves to a String value: the authed userId
    */
-  authenticate(providerType, options = {}) {
+  authenticate(providerType, options = {}, link = false) {
     // reuse existing auth if present
     const authenticateFn = () =>
-      this.auth.provider(providerType).authenticate(options).then(() => this.authedId());
+      this.auth.provider(providerType).authenticate(options, link).then(() => this.authedId());
 
     if (this.isAuthenticated()) {
       if (providerType === PROVIDER_TYPE_ANON && this.auth.getLoggedInProviderType() === PROVIDER_TYPE_ANON) {
         return Promise.resolve(this.auth.authedId); // is authenticated, skip log in
       }
 
+      if (link) {
```

Comment:
[minor] I believe this belongs outside of the isAuthenticated block. The logic is, is this link? if yes, throw if not authenticated; otherwise call `authenticateFn`. If no, continue on with existing code. Can you add a test case that we throw when you try to link without being authed?